var/spack/repos/builtin/packages/libfabric/package.py

```diff
@@ -89,6 +94,12 @@ class Libfabric(AutotoolsPackage):
 
     variant("debug", default=False, description="Enable debugging")
 
+    variant(
```

Comment:
Good idea. I didn't realize how few providers support fabric direct. From a quick look through the libfabric repo, it looks like the providers that _do support_ this feature are: bgq, psm2, opx, and sockets. Does this look complete and correct to you, @rajachan?

Reply:
I moved this to a draft because it looks like libfabric is having trouble finding various files required for direct at configure time. This is really strange to me. 

Taking PSM2 as an example, if I look at the provider in the 1.17.1 tagged version,[ the `rdma` subdirectory is present](https://github.com/ofiwg/libfabric/tree/v1.17.1/prov/psm2/include/rdma). After downloading and unpacking the release tarball, it's missing. I assume that's why Spack is tripping over it:
```
1 error found in build log:
     164    checking for psm2.h... yes
     165    configure: looking for library without search path
     166    checking for psm2_info_query in -lpsm2... yes
     167    configure: psm2 provider: include in libfabric
     168    checking for ./prov/psm2/include/rdma/fi_direct.h... no
     169    checking for ./prov/psm2/include/rdma/fi_direct.h.in... no
  >> 170    configure: error: psm2 provider was requested as direct, but is missing fi_direct.h and fi_direct.h.in
```

Similarly for OPX,[ the file it can't find seems to exist](https://github.com/ofiwg/libfabric/blob/v1.17.1/prov/opx/provider_FABRIC_1.0.map) 🤔 :
```
1 error found in build log:
     222    checking rdma/hfi/hfi1_user.h presence... yes
     223    checking for rdma/hfi/hfi1_user.h... yes
     224    checking whether HAVE_ATOMICS is declared... yes
     225    configure: opx provider: include in libfabric
     226    checking for ./prov/opx/include/rdma/fi_direct.h... yes
     227    configure: WARNING: --enable-direct=opx specified, but ./prov/opx/provider_FABRIC_1.0.map does not exist
  >> 228    configure: error: Cannot continue
```

@rajachan do you have any ideas why this might be?

Reply:
Hmm, ok. Thanks for checking. I'll try to figure out why I'm not seeing this behavior through Spack at this PR.